### DISCUSSION TITLE
Multiple characters, but one session max?

### DISCUSSION THREAD

**Discussion by owllex** [evennia/evennia#2523] [Evennia Questions] [answered]
I'd like to implement a policy of supporting multiple characters per account but only allowing one character to be played at a time (except developers). This seems like it's closest to `MULTISESSION_MODE=2`, but with some additional restrictions.

The motivation for this is that I'd like to be able to keep characters grouped into a single account for convenience, but I don't want multiplaying to be a thing.

I imagine I'd need to check for an existing connected session after a new session logs in and disconnect the old one (based on permissions). Where would be the right place to put this logic? Or is there a better way to accomplish this?

Thanks in advance!

**Comment by kovitikus** [comment, marked as answer]
Your best approach is to use multisession mode 1:
`multiple sessions, one account, one character, each session getting the same data`

and then simply take care of character management yourself. It's a hell of a lot easier to do this than to change the session behavior, IMO.

You have to make your own character generation, store those characters to the player's account, and then develop a menu that lets players manage the characters (play/delete). This way they can only puppet one at a time.

It's the approach I'm taking, at least.

**Reply by owllex** [reply]
Thanks for the advice. I tried this out but what I got stuck on was the fact that MULTISESSION_MODE=1 causes the first character to be automatically created with the same name as the account. I didn't want to re-write session logic to get around this.

I ended up doing mostly what you suggested, but I set the game to MULTISESSION_MODE=2. This way, a new account is created but the user's first character can be whatever they want (useful since account and character names have different requirements).

I wrote my own character create/delete commands, and built my menu-driven character creator into that create command. Works great now!

**Comment by Belameephus** [comment]
I don't have an answer for the question owllex posted. I have been struggling with getting my game to work as owllex says he wants (or hopefully has been able to get) his game to work: MultiSession per Account, multiple Characters created per Account, ONLY one Character puppeted by an Account at any one time, logon takes new Account to an Account Menu, takes existing Accounts to last puppet if one was puppeted otherwise to Account MEnu! I have most logic figured out for all things IC, it is just the logging on as I need that has me stumped (been struggling with this for over 3 years. I am not an experienced Python, nor object oriented, programmer/developer. Would someone be willing to give explicit examples of how to accomplish what I am trying to do, preferably with example. Thanks in advance for any help (BTW if you live in my area, I may be willing to pay to get past this issue.

**Reply by Griatch** [reply]
@Belameephus The logging-on behavior is pretty much all in the register/connect unloggedin-commands. You can override these as you please (same as any commands, just that you override in the UnloggedinCmdSet instead). Here's the `connect` command for reference: https://github.com/evennia/evennia/blob/master/evennia/commands/default/unloggedin.py#L87

If you have more direct questions, it may be better to use a separate discussion thread for that, or to pop into the support chat where you can ask and get more quicker, specific answers as you go.

**Reply by Belameephus** [reply]
Sorry, my mistake. 
I should have started my question as a new thread, again.
The ununloggedin connect command actually calls the session handler to make the connection, similar to how the unloggedin create command calls account.create. 

If I want MULTISESSION_MODE = 1 but do not want a Character auto-created, that is done elsewhere.
Once again, sorry. My mistake for not being a bit more clear with explaining my problem. 
If I cannot figure out what causes my game to lock up, preventing any login at the moment, I will again start over with an new install, then ask elsewhere for assistance.

Thanks for any help provided. Griatch, you and the whole Evennia team and community have done an excellent job putting together what may become the best MUD game creation system yet. I really do appreciate what you have done. It gives me hope that my 40+ years of designing a roleplay game world may eventually produce an online game world that others can experience.

If I sound a bit 'grouchy', it is my fault - too many hours without sleep with yet another mistake on my part, and time is running out for me. I apologize. 

Please, keep up the good work. For now, I must return to my programming project.
